src/App.tsx

```diff
@@ -20,17 +25,126 @@ class App extends React.Component<{}, State> {
       './img/9.png',
       './img/10.png',
     ],
+    itemWidth: 130,
+    frameSize: 3,
+    step: 3,
+    animationDuration: 1000,
+    infinite: false,
   };
 
+  handleItemWidthChange(event: React.ChangeEvent<HTMLInputElement>) {
+    this.setState({ itemWidth: +event.target.value });
+  }
+
+  handleFrameSizeChange(event: React.ChangeEvent<HTMLInputElement>) {
+    this.setState({ frameSize: +event.target.value });
+  }
+
+  handleStepChange(event: React.ChangeEvent<HTMLInputElement>) {
+    this.setState({ step: +event.target.value });
+  }
+
+  handleDurationChange(event: React.ChangeEvent<HTMLInputElement>) {
+    this.setState({ animationDuration: +event.target.value });
+  }
+
+  handleInfiniteChange(event: React.ChangeEvent<HTMLInputElement>) {
+    this.setState({ infinite: event.target.checked });
+  }
+
   render() {
-    const { images } = this.state;
+    const {
+      images,
+      itemWidth,
+      frameSize,
+      step,
+      animationDuration,
+      infinite,
+    } = this.state;
 
     return (
       <div className="App">
+
         {/* eslint-disable-next-line */}
-        <h1>Carousel with {images.length} images</h1>
+        <h1 className="title" data-cy="title">Carousel with {images.length} images</h1>
+
+        <form
+          className="settings"
+          action="#"
+          method="get"
+        >
+          <label className="settings__title" htmlFor="itemId">
+            Image size:
+            <input
+              className="settings__input"
+              type="number"
+              name="itemWidth"
+              id="itemId"
+              step={10}
+              value={itemWidth}
+              onChange={(event) => this.handleItemWidthChange(event)}
```

Comment:
```suggestion
              onChange={this.handleItemWidthChange}
```
**PLS DO THE SAME FOR ALL SUCH CASES IN THE PROJECT**